pytket/extensions/cutensornet/general_state/tensor_network_state.py

```diff
@@ -0,0 +1,466 @@
+        self._work_desc = None
+
+        self._state = cutn.create_state(
+            self._handle, cutn.StatePurity.PURE, num_qubits, qubits_dims, data_type
+        )
+        self._mutable_gates_map = {}
+        self._gate_tensors = []
+        for com in circuit.get_commands():
+            gate_unitary = com.op.get_unitary().astype("complex128", copy=False)
+            # Transpose is needed because of the way cuTN stores tensors.
+            # See https://github.com/NVIDIA/cuQuantum/discussions/124
+            # #discussioncomment-8683146 for details.
+            self._gate_tensors.append(
+                cp.asarray(gate_unitary)
+                .T.astype(dtype="complex128", order="F")
+                .reshape([2] * (2 * com.op.n_qubits), order="F")
+            )
+            gate_strides = 0  # Always 0?
+            gate_qubit_indices = tuple(
+                self._circuit.qubits.index(qb) for qb in com.qubits
+            )
+            gate_id = cutn.state_apply_tensor(
+                self._handle,
+                self._state,
+                com.op.n_qubits,
+                gate_qubit_indices,
+                self._gate_tensors[-1].data.ptr,
+                gate_strides,
+                1,
+                0,
+                1,
+            )
+            if com.opgroup is not None:
+                self._mutable_gates_map[com.opgroup] = gate_id
+
+    @property
+    def state(self) -> Any:
+        """Returns tensor network state handle as Python :code:`int`."""
+        return self._state
+
+    def update_gates(self, gates_update_map: dict[str, NDArray]) -> None:
+        """Updates gate unitaries in the tensor network state.
+
+        Args:
+            gates_update_map: Map from gate (Command) opgroup name to a corresponding
+             gate unitary.
+
+        Raises:
+            ValueError: If a gate's unitary is of a wrong size.
+        """
+        for gate_label, unitary in gates_update_map.items():
+            gate_id = self._mutable_gates_map[gate_label]
+            gate_n_qubits = math.log2(unitary.shape[0])
+            if not gate_n_qubits.is_integer():
+                raise ValueError(
+                    f"Gate {gate_label} unitary's number of rows is not a power of two."
+                )
+            gate_tensor = cp.asarray(unitary, dtype="complex128").reshape(
+                [2] * (2 * int(gate_n_qubits)), order="F"
+            )
+            cutn.state_update_tensor(
+                self._handle, self._state, gate_id, gate_tensor.data.ptr, 1
+            )
+
+    def configure(self, attributes: Optional[dict] = None) -> GeneralState:
+        """Configures tensor network state for future contraction.
+
+        Args:
+            attributes: A dict of cuTensorNet State attributes and their values.
+
+        Returns:
+            Self (to allow for chaining with other methods).
+        """
+        if attributes is None:
+            attributes = {"NUM_HYPER_SAMPLES": 8}
```

Comment:
Does it work without providing an explicit one? If so, that'd mean that NVIDIA has their own default choice, why would we hardcode a different one?

Reply:
It works without setting a default value. I've removed it, since I'd much rather it default to whatever NVIDIA chose, and it's bad practice to set default values for parameters without informing the user.

@yapolyak let me know if there's a strong reason to set this value to 8.